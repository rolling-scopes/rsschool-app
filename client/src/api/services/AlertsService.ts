/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */
import type { CreateAlertDto } from '../models/CreateAlertDto';
import type { UpdateAlertDto } from '../models/UpdateAlertDto';
import type { CancelablePromise } from '../core/CancelablePromise';
import { request as __request } from '../core/request';

export class AlertsService {

    /**
     * @param requestBody
     * @returns any
     * @throws ApiError
     */
    public static alertsControllerCreate(
        requestBody: CreateAlertDto,
    ): CancelablePromise<any> {
        return __request({
            method: 'POST',
            path: `/api/v2/alerts`,
            body: requestBody,
            mediaType: 'application/json',
        });
    }

    /**
     * @param enabled
     * @returns any
     * @throws ApiError
     */
    public static alertsControllerGetAll(
        enabled: boolean,
    ): CancelablePromise<any> {
        return __request({
            method: 'GET',
            path: `/api/v2/alerts`,
            query: {
                'enabled': enabled,
            },
        });
    }

    /**
     * @param id
     * @param requestBody
     * @returns any
     * @throws ApiError
     */
    public static alertsControllerUpdate(
        id: number,
        requestBody: UpdateAlertDto,
    ): CancelablePromise<any> {
        return __request({
            method: 'PATCH',
            path: `/api/v2/alerts/${id}`,
            body: requestBody,
            mediaType: 'application/json',
        });
    }

    /**
     * @param id
     * @returns any
     * @throws ApiError
     */
    public static alertsControllerRemove(
        id: number,
    ): CancelablePromise<any> {
        return __request({
            method: 'DELETE',
            path: `/api/v2/alerts/${id}`,
        });
    }

}