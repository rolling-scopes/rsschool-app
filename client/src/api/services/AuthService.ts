/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */
import type { CancelablePromise } from '../core/CancelablePromise';
import { request as __request } from '../core/request';

export class AuthService {

    /**
     * @returns any
     * @throws ApiError
     */
    public static authControllerGithubLogin(): CancelablePromise<any> {
        return __request({
            method: 'GET',
            path: `/api/v2/auth/github/login`,
        });
    }

    /**
     * @returns any
     * @throws ApiError
     */
    public static authControllerGithubCallback(): CancelablePromise<any> {
        return __request({
            method: 'GET',
            path: `/api/v2/auth/github/callback`,
        });
    }

}