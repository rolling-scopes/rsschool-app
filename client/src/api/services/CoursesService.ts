/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */
import type { CancelablePromise } from '../core/CancelablePromise';
import { request as __request } from '../core/request';

export class CoursesService {

    /**
     * @returns any
     * @throws ApiError
     */
    public static coursesControllerGetCourses(): CancelablePromise<any> {
        return __request({
            method: 'GET',
            path: `/api/v2/courses`,
        });
    }

    /**
     * @param aliasOrId
     * @returns any
     * @throws ApiError
     */
    public static getCourse(
        aliasOrId: string,
    ): CancelablePromise<any> {
        return __request({
            method: 'GET',
            path: `/api/v2/courses/${aliasOrId}`,
        });
    }

}