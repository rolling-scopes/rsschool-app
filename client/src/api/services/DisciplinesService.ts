/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */
import type { CreateDisciplineDto } from '../models/CreateDisciplineDto';
import type { UpdateDisciplineDto } from '../models/UpdateDisciplineDto';
import type { CancelablePromise } from '../core/CancelablePromise';
import { request as __request } from '../core/request';

export class DisciplinesService {

    /**
     * @param requestBody
     * @returns any
     * @throws ApiError
     */
    public static disciplinesControllerCreate(
        requestBody: CreateDisciplineDto,
    ): CancelablePromise<any> {
        return __request({
            method: 'POST',
            path: `/api/v2/disciplines`,
            body: requestBody,
            mediaType: 'application/json',
        });
    }

    /**
     * @returns any
     * @throws ApiError
     */
    public static disciplinesControllerGetAll(): CancelablePromise<any> {
        return __request({
            method: 'GET',
            path: `/api/v2/disciplines`,
        });
    }

    /**
     * @param id
     * @returns any
     * @throws ApiError
     */
    public static disciplinesControllerDelete(
        id: number,
    ): CancelablePromise<any> {
        return __request({
            method: 'DELETE',
            path: `/api/v2/disciplines/${id}`,
        });
    }

    /**
     * @param id
     * @param requestBody
     * @returns any
     * @throws ApiError
     */
    public static disciplinesControllerUpdate(
        id: number,
        requestBody: UpdateDisciplineDto,
    ): CancelablePromise<any> {
        return __request({
            method: 'PATCH',
            path: `/api/v2/disciplines/${id}`,
            body: requestBody,
            mediaType: 'application/json',
        });
    }

}