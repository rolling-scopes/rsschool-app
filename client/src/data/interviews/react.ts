import { InterviewTemplate } from './types';

export const reactTemplate: InterviewTemplate = {
  name: 'React interview',
  examplesUrl: 'https://github.com/rolling-scopes-school/tasks/blob/master/react/questions.md',
  categories: [
    {
      id: 1000,
      name: 'JSX',
      questions: [
        { id: 1001, name: 'What is JSX?' },
        { id: 1002, name: 'Is it possible to use React without JSX?' },
      ],
    },
    {
      id: 1010,
      name: 'Styled-components',
      questions: [{ id: 1011, name: 'Using styled-components' }],
    },
    {
      id: 1020,
      name: 'Components',
      questions: [
        { id: 1021, name: 'Component Definition (Class-based, Functional)' },
        { id: 1022, name: 'Components Composition' },
      ],
    },
    {
      id: 1030,
      name: 'React DOM',
      questions: [
        { id: 1031, name: 'What is the virtual DOM? How does react use the virtual DOM to render the UI?' },
        { id: 1032, name: 'Is the virtual DOM the same as the shadow DOM?' },
        { id: 1033, name: 'What is the difference between the virtual DOM and the real DOM?' },
      ],
    },
    {
      id: 1040,
      name: 'Render',
      questions: [
        { id: 1041, name: 'When is a component rendered?' },
        { id: 1042, name: 'How not to render on props change?' },
        { id: 1043, name: 'Is it OK to use arrow functions in render methods?' },
        { id: 1044, name: 'Components rendering system' },
      ],
    },
    {
      id: 1050,
      name: 'Interaction between components',
      questions: [
        { id: 1051, name: 'How do you pass a value from parent to child?' },
        { id: 1052, name: 'How do you pass a value from child to parent?' },
        { id: 1053, name: 'What is prop drilling?' },
        { id: 1054, name: 'Can a child component modify its own props?' },
        { id: 1055, name: 'How do you pass a value from sibling to sibling?' },
      ],
    },
    {
      id: 1060,
      name: 'Lifecycle and State',
      questions: [
        { id: 1061, name: 'What is the difference between props and state?' },
        { id: 1062, name: 'How does state in a class component differ from state in a functional component?' },
        { id: 1063, name: 'What is the component lifecycle?' },
        { id: 1064, name: 'How do you update lifecycle in function components?' },
        { id: 1065, name: 'Controlled/uncontrolled components' },
        { id: 1066, name: 'Stateful vs stateless components' },
      ],
    },
    {
      id: 1070,
      name: 'Ref',
      questions: [
        { id: 1071, name: 'What is the difference between refs and state variables?' },
        { id: 1072, name: 'When is the best time to use refs?' },
        { id: 1073, name: 'What is the proper way to update a ref in a function component?' },
      ],
    },
    {
      id: 1080,
      name: 'Context',
      questions: [
        { id: 1081, name: 'What is the difference between the context API and prop drilling?' },
        { id: 1082, name: "When shouldn't you use the context API?" },
      ],
    },
    {
      id: 1090,
      name: 'Redux',
      questions: [
        { id: 1091, name: 'Enumerate base principles' },
        { id: 1092, name: 'What is the typical flow of data in a React + Redux app?' },
        { id: 1093, name: 'Benefits of Redux?' },
        { id: 1094, name: 'Async Redux flow' },
      ],
    },
    {
      id: 1100,
      name: 'State Management by managers',
      questions: [
        { id: 1101, name: 'Mobx' },
        { id: 1102, name: 'Reflux' },
      ],
    },
    {
      id: 1110,
      name: 'Routing',
      questions: [
        { id: 1111, name: 'React Router' },
        { id: 1112, name: 'History' },
      ],
    },
    {
      id: 1120,
      name: 'Other',
      questions: [
        { id: 1121, name: 'Is it a good idea to use Math.random for keys?' },
        { id: 1122, name: 'What are the limitations of React?' },
        { id: 1123, name: 'What is a higher order component?' },
        { id: 1124, name: 'What are uncontrolled and controlled components?' },
        { id: 1125, name: 'React optimizations' },
      ],
    },
    {
      id: 1130,
      name: 'Coding task',
      questions: [{ id: 1131, name: 'Small react app: form, button, results list' }],
    },
    {
      id: 1140,
      name: 'Next.js',
      questions: [
        { id: 1141, name: 'What is Next.js, and how does it differ from traditional React applications?' },
        { id: 1142, name: 'Explain the concept of server-side rendering (SSR) in Next.js' },
        { id: 1143, name: 'How does Next.js handle client-side routing?' },
        { id: 1144, name: 'What are the benefits of using Next.js for building React applications?' },
        {
          id: 1145,
          name: 'What is the purpose of the getInitialProps function in Next.js? How is it different from using getStaticProps or getServerSideProps?',
        },
        { id: 1146, name: 'How does Next.js handle automatic code splitting, and why is it important?' },
        {
          id: 1147,
          name: 'What is the purpose of the Link component in Next.js, and how does it differ from traditional anchor (<a>) tags?',
        },
        { id: 1148, name: 'How can you configure routing in a Next.js application?' },
      ],
    },
    {
      id: 1150,
      name: 'Data Fetching',
      questions: [
        {
          id: 1151,
          name: 'Compare and contrast getStaticProps and getServerSideProps in terms of use cases and performance considerations',
        },
        {
          id: 1152,
          name: 'When would you choose to use getStaticPaths in Next.js, and how does it relate to dynamic routes?',
        },
      ],
    },
    {
      id: 1160,
      name: 'Advanced Topics',
      questions: [
        { id: 1161, name: 'Describe the purpose and use cases of API routes in Next.js' },
        { id: 1162, name: 'Explain the concept of middleware in Next.js and its role in the application lifecycle' },
        { id: 1163, name: 'How does Next.js handle authentication in applications?' },
      ],
    },
  ],
};
