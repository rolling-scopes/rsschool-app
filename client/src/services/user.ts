import axios, { AxiosInstance } from 'axios';
import { NextPageContext, GetServerSidePropsContext } from 'next';
import { getServerAxiosProps } from 'utils/axios';
import { EnglishLevel } from '../../../common/models';
import { ProfileInfo, SaveProfileInfo } from '../../../common/models/profile';
import { Course } from './models';
import discordIntegration from '../configs/discord-integration';

export interface UserBasic {
  name: string;
  githubId: string;
  id: number;
}

type SearchResponse = { data: UserBasic[] };

export class UserService {
  private axios: AxiosInstance;

  constructor(ctx?: GetServerSidePropsContext | NextPageContext) {
    this.axios = axios.create(getServerAxiosProps(ctx));
  }

  async getDiscordIds() {
    const fragment = new URLSearchParams(window.location.hash.slice(1));

    if (fragment.has('access_token')) {
      const accessToken = fragment.get('access_token');
      const tokenType = fragment.get('token_type');

      const response = await fetch(discordIntegration.api.me, {
        headers: {
          authorization: `${tokenType} ${accessToken}`,
        },
      });

      const { username, discriminator, id } = await response.json();

      return {
        username,
        discriminator,
        id,
      };
    }

    return null;
  }

  async getCourses() {
    const result = await this.axios.get<{ data: Course[] }>(`/api/user/me/courses`);
    return result.data.data.sort((a, b) => b.id - a.id);
  }

  async searchUser(query: string | null) {
    try {
      if (!query) {
        return [];
      }
      const response = await this.axios.get<SearchResponse>(`/api/users/search/${query}`);
      return response.data.data;
    } catch (e) {
      return [];
    }
  }

  async extendedUserSearch(query: string | null) {
    try {
      if (!query) {
        return [];
      }
      const response = await this.axios.get<SearchResponse>(`/api/users/search/extended/${query}`);
      return response.data.data;
    } catch (e) {
      return [];
    }
  }

  async submitPrivateFeedback(data: { toUserId: number; comment: string }) {
    await this.axios.post(`/api/feedback/private`, {
      toUserId: Number(data.toUserId),
      comment: data.comment,
    });
  }

  async submitStudentFeedback(data: { toUserId: number; comment: string }, courseId: string) {
    await this.axios.post(`api/course/${courseId}/student/feedback`, {
      toUserId: Number(data.toUserId),
      comment: data.comment,
    });
  }

  async getMyProfile() {
    const response = await this.axios.get<{ data: UserFull }>(`/api/profile/me`);
    return response.data.data;
  }

  async updateMyProfile(data: Partial<UserFull>) {
    const response = await this.axios.post<{ data: UserFull }>(`/api/profile/me`, data);
    return response.data.data;
  }

  async getProfileInfo(githubId?: string) {
    const response = await this.axios.get<{ data: ProfileInfo }>(`/api/profile/info`, {
      params: { githubId },
    });
    return response.data.data;
  }

  async saveProfileInfo(profile: SaveProfileInfo) {
    const response = await this.axios.post<{ data: SaveProfileInfo }>(`/api/profile/info`, profile);
    return response.data.data;
  }
}

export type ResponseStudent = {
  id: number;
  totalScore: number;
  certificatePublicId: string;
  completed: boolean;
  stageInterviews: {
    date: string;
    isGoodCandidate: boolean;
    english: EnglishLevel;
    comment: string;
    rating: number;
    programmingTask: {
      task: string;
      resolved: number;
      codeWritingLevel: number;
      comment: string;
    };
    interviewer: {
      githubId: string;
      name: string;
    };
    skills: {
      htmlCss: number;
      common: number;
      dataStructures: number;
    };
  }[];
  interviews: {
    score: number;
    comment: string;
    formAnswers: {
      questionText: string;
      answer: string;
    }[];
    courseTask: {
      id: number;
      name: string;
      descriptionUrl: string;
    };
  }[];
  taskResults: {
    score: number;
    githubPrUrl: string;
    comment: string;
    courseTask: {
      id: number;
      name: string;
      descriptionUrl: string;
    };
  }[];
  mentor: {
    id: number;
    githubId: string;
    name: string;
  } | null;
};

export type ResponseMentor = {
  id: number;
  students: {
    id: number;
    userId: number;
    githubId: string;
    lastName: string;
    firstName: string;
  }[];
};

export type ResponseCourse = {
  id: number;
  name: string;
};

export interface UserFull extends UserBasic {
  firstName: string;
  lastName: string;
  externalAccounts: any[];
  englishLevel: string;
  readyFullTime: string;
  educationHistory: any[];
  employmentHistory: any[];
  contactsTelegram: string;
  contactsSkype: string;
  contactsEmail: string;
  primaryEmail: string;
  contactsEpamEmail: string;
  contactsPhone: string;
  contactsNotes: string;
  locationId: number;
  locationName: string;
  aboutMyself: string;
  tshirtSize: string;
  countryName: string;
  cityName: string;
}

export interface ProfileResponse {
  user: UserFull;
  students: (ResponseStudent & { course: ResponseCourse })[];
  mentors: (ResponseMentor & { course: ResponseCourse })[];
}
