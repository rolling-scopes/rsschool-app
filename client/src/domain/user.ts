import { values } from 'lodash';
import { Session } from 'components/withSession';
import { Course } from 'services/models';

export function isAdmin(session: Session) {
  return session.isAdmin;
}

export function isMentor(session: Session, course: Course) {
  return session.roles?.[course.id] === 'mentor';
}

export function isStudent(session: Session, course: Course) {
  return session.roles?.[course.id] === 'student';
}

export function isCourseManager(session: Session, course: Course) {
  return session.isAdmin || (session.coursesRoles?.[course.id]?.includes('manager') ?? false);
}

export function isAnyCourseManager(session: Session) {
  return values(session.coursesRoles).some(item => item?.includes('manager'));
}

export function isCourseSupervisor(session: Session, course: Course) {
  return session.isAdmin || (session.coursesRoles?.[course.id]?.includes('supervisor') ?? false);
}

export function isAnyCourseSupervisor(session: Session) {
  return values(session.coursesRoles).some(item => item?.includes('supervisor'));
}

export function isAnyCoursePowerUserManager(session: Session) {
  return isAnyCourseManager(session) || isAnyCourseSupervisor(session);
}

export function isTaskOwner(session: Session, course: Course) {
  return session.coursesRoles?.[course.id]?.includes('taskOwner') ?? false;
}

export function isJuryActivist(session: Session, course: Course) {
  return session.coursesRoles?.[course.id]?.includes('juryActivist') ?? false;
}
