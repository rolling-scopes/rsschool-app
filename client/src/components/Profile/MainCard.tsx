import * as React from 'react';
import isEqual from 'lodash/isEqual';
import { GithubAvatar } from 'components/GithubAvatar';
import { Card, Typography, Input, Row, Col } from 'antd';
import { GeneralInfo, ConfigurableProfilePermissions } from '../../../../common/models/profile';
import { ChangedPermissionsSettings } from 'pages/profile';
import { CheckboxChangeEvent } from 'antd/lib/checkbox';
import { LocationSelect } from 'components/Forms';
import PermissionsSettingsDrawer from './PermissionsSettingsDrawer';
import ProfileSettingsDrawer from './ProfileSettingsDrawer';

const { Title, Paragraph, Text } = Typography;

import { GithubFilled, EnvironmentFilled, EditOutlined, SettingOutlined } from '@ant-design/icons';

type Props = {
  data: GeneralInfo;
  isEditingModeEnabled: boolean;
  permissionsSettings?: ConfigurableProfilePermissions;
  onPermissionsSettingsChange: (event: CheckboxChangeEvent, settings: ChangedPermissionsSettings) => void;
  onProfileSettingsChange: (event: any, path: string) => void;
};

type State = {
  isVisibilitySettingsVisible: boolean;
  isProfileSettingsVisible: boolean;
};

class MainCard extends React.Component<Props, State> {
  state: State = {
    isVisibilitySettingsVisible: false,
    isProfileSettingsVisible: false,
  };

  shouldComponentUpdate = (nextProps: Props, nextState: State) =>
    !isEqual(nextProps.data.location.cityName, this.props.data.location.cityName) ||
    !isEqual(nextProps.data.location.countryName, this.props.data.location.countryName) ||
    !isEqual(nextProps.data.name, this.props.data.name) ||
    !isEqual(nextProps.permissionsSettings?.isProfileVisible, this.props.permissionsSettings?.isProfileVisible) ||
    !isEqual(nextProps.isEditingModeEnabled, this.props.isEditingModeEnabled) ||
    !isEqual(nextState, this.state);

  private showVisibilitySettings = () => {
    this.setState({ isVisibilitySettingsVisible: true });
  };

  private hideVisibilitySettings = () => {
    this.setState({ isVisibilitySettingsVisible: false });
  };

  private showProfileSettings = () => {
    this.setState({ isProfileSettingsVisible: true });
  };

  private hideProfileSettings = () => {
    this.setState({ isProfileSettingsVisible: false });
  };
  private filterPermissions = ({ isProfileVisible }: Partial<ConfigurableProfilePermissions>) => ({
    isProfileVisible,
  });

  render() {
    const { isEditingModeEnabled, permissionsSettings, onPermissionsSettingsChange, onProfileSettingsChange } =
      this.props;
    const { githubId, name, location } = this.props.data;
    const { isProfileSettingsVisible, isVisibilitySettingsVisible } = this.state;

    return (
      <>
        <Card
          actions={
            isEditingModeEnabled
              ? [
                  <EditOutlined key="main-card-actions-edit" onClick={this.showProfileSettings} />,
                  <SettingOutlined key="main-card-actions-settings" onClick={this.showVisibilitySettings} />,
                ]
              : undefined
          }
        >
          <GithubAvatar size={96} githubId={githubId} style={{ margin: '0 auto 10px', display: 'block' }} />
          <Title level={1} style={{ fontSize: 24, textAlign: 'center', margin: 0 }}>
            {name}
          </Title>
          <Paragraph style={{ textAlign: 'center', marginBottom: 20 }}>
            <a target="_blank" href={`https://github.com/${githubId}`} style={{ marginLeft: '-14px', fontSize: 16 }}>
              <GithubFilled /> {githubId}
            </a>
          </Paragraph>
          <Paragraph style={{ textAlign: 'center', margin: 0 }}>
            <span style={{ marginLeft: '-14px' }}>
              <EnvironmentFilled /> {`${location.cityName}, ${location.countryName}`}
            </span>
          </Paragraph>
          {isEditingModeEnabled && (
            <>
              <PermissionsSettingsDrawer
                isSettingsVisible={isVisibilitySettingsVisible}
                hideSettings={this.hideVisibilitySettings}
                permissionsSettings={permissionsSettings ? this.filterPermissions(permissionsSettings) : undefined}
                onPermissionsSettingsChange={onPermissionsSettingsChange}
              />
              <ProfileSettingsDrawer
                isSettingsVisible={isProfileSettingsVisible}
                hideSettings={this.hideProfileSettings}
                content={
                  <Row>
                    <Col>
                      <Row>
                        <Text strong>Name</Text>
                      </Row>
                      <Row style={{ marginTop: 4 }}>
                        <Input
                          value={name}
                          placeholder="Firstname Lastname"
                          onChange={(event: any) => onProfileSettingsChange(event, 'generalInfo.name')}
                        />
                      </Row>

                      <Row style={{ marginTop: 24 }}>
                        <Text strong>Location</Text>
                      </Row>
                      <Row style={{ marginTop: 4 }}>
                        <LocationSelect
                          style={{ flex: 1 }}
                          onChange={location => onProfileSettingsChange(location, 'generalInfo.location')}
                          location={location}
                        />
                      </Row>
                    </Col>
                  </Row>
                }
              />
            </>
          )}
        </Card>
      </>
    );
  }
}

export default MainCard;
