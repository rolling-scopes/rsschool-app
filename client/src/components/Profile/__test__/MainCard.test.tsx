import React from 'react';
import { render } from '@testing-library/react';
import MainCard from '../MainCard';
import { ProfileInfo } from 'services/user';

// TODO: Known Issue: https://stackoverflow.com/questions/59942808/how-can-i-use-jest-coverage-in-next-js-styled-jsx

const mockData = {
  generalInfo: {
    name: 'John Doe',
    githubId: 'john-doe',
    location: {
      countryName: 'Belarus',
      cityName: 'Minsk',
    },
  },
  publicCvUrl: 'public-url',
} as ProfileInfo;

describe('MainCard', () => {
  describe('Should render correctly', () => {
    it('if is editing mode disabled', () => {
      const output = render(
        <MainCard
          data={mockData}
          isEditingModeEnabled={false}
          onPermissionsSettingsChange={jest.fn()}
          onProfileSettingsChange={jest.fn()}
        />,
      );
      expect(output.container).toMatchSnapshot();
    });
    it('if is editing mode enabled', () => {
      const output = render(
        <MainCard
          data={mockData}
          isEditingModeEnabled={true}
          onPermissionsSettingsChange={jest.fn()}
          onProfileSettingsChange={jest.fn()}
        />,
      );
      expect(output.container).toMatchSnapshot();
    });
  });
});
