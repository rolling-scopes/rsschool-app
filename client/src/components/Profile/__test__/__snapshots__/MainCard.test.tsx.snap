// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainCard Should render correctly if is editing mode disabled 1`] = `
<Fragment>
  <Card>
    <GithubAvatar
      githubId="piter"
      size={96}
      style={
        Object {
          "display": "block",
          "margin": "0 auto 10px",
        }
      }
    />
    <Title
      level={1}
      style={
        Object {
          "fontSize": 24,
          "margin": 0,
          "textAlign": "center",
        }
      }
    >
      Petr Pervyi
    </Title>
    <Paragraph
      style={
        Object {
          "marginBottom": 20,
          "textAlign": "center",
        }
      }
    >
      <a
        href="https://github.com/piter"
        style={
          Object {
            "fontSize": 16,
            "marginLeft": "-14px",
          }
        }
        target="_blank"
      >
        <ForwardRef(GithubFilled) />
         
        piter
      </a>
    </Paragraph>
    <Paragraph
      style={
        Object {
          "margin": 0,
          "textAlign": "center",
        }
      }
    >
      <span
        style={
          Object {
            "marginLeft": "-14px",
          }
        }
      >
        <ForwardRef(EnvironmentFilled) />
         
        Minsk, Belarus
      </span>
    </Paragraph>
  </Card>
</Fragment>
`;

exports[`MainCard Should render correctly if is editing mode enabled 1`] = `
<Fragment>
  <Card
    actions={
      Array [
        <ForwardRef(EditOutlined)
          onClick={[Function]}
        />,
        <ForwardRef(SettingOutlined)
          onClick={[Function]}
        />,
      ]
    }
  >
    <GithubAvatar
      githubId="piter"
      size={96}
      style={
        Object {
          "display": "block",
          "margin": "0 auto 10px",
        }
      }
    />
    <Title
      level={1}
      style={
        Object {
          "fontSize": 24,
          "margin": 0,
          "textAlign": "center",
        }
      }
    >
      Petr Pervyi
    </Title>
    <Paragraph
      style={
        Object {
          "marginBottom": 20,
          "textAlign": "center",
        }
      }
    >
      <a
        href="https://github.com/piter"
        style={
          Object {
            "fontSize": 16,
            "marginLeft": "-14px",
          }
        }
        target="_blank"
      >
        <ForwardRef(GithubFilled) />
         
        piter
      </a>
    </Paragraph>
    <Paragraph
      style={
        Object {
          "margin": 0,
          "textAlign": "center",
        }
      }
    >
      <span
        style={
          Object {
            "marginLeft": "-14px",
          }
        }
      >
        <ForwardRef(EnvironmentFilled) />
         
        Minsk, Belarus
      </span>
    </Paragraph>
    <PermissionsSettingsDrawer
      hideSettings={[Function]}
      isSettingsVisible={false}
      onPermissionsSettingsChange={[MockFunction]}
    />
    <ProfileSettingsDrawer
      content={
        <Row>
          <Col>
            <Row>
              <Text
                strong={true}
              >
                Name
              </Text>
            </Row>
            <Row
              style={
                Object {
                  "marginTop": 4,
                }
              }
            >
              <Input
                onChange={[Function]}
                placeholder="Firstname Lastname"
                type="text"
                value="Petr Pervyi"
              />
            </Row>
            <Row
              style={
                Object {
                  "marginTop": 24,
                }
              }
            >
              <Text
                strong={true}
              >
                Location
              </Text>
            </Row>
            <Row
              style={
                Object {
                  "marginTop": 4,
                }
              }
            >
              <LocationSelect
                location={
                  Object {
                    "cityName": "Minsk",
                    "countryName": "Belarus",
                  }
                }
                onChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                  }
                }
              />
            </Row>
          </Col>
        </Row>
      }
      hideSettings={[Function]}
      isSettingsVisible={false}
    />
  </Card>
</Fragment>
`;
